### PR TITLE
plugins: add markunmatched

### PR DESCRIPTION
**Pull request by doronbehar**
## Description

Add a plugin useful for bulk importers.

## To Do

- [x] Documentation. (If you've add a new command-line flag, for example, find the appropriate page under `docs/` to describe it.)
- [x] Changelog. (Add an entry to `docs/changelog.rst` near the top of the document.)
- [ ] Tests. (Encouraged but not strictly required.)

### CODE REVIEW

File: beetsplug/markunmatched.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                print(stdout.decode('utf-8'), end='')
+        return [PromptChoice(
+            'f',
+            'create a \'beets-unmatched\' File and skip',
+            self.touch_unmatched
+        )]
+
+    def touch_unmatched(self, session, task):
+        for path in task.paths:
+            Path(path.decode('utf-8') + "/beets-unmatched").touch()
```
Comment (ybnd) [inline beetsplug/markunmatched.py:52]: I imagine you wanted this file to be easily visible from the terminal/file browser, but my first thought was that it should probably be hidden (i.e. `.beets-unmatched`).

Could be useful to make the filename configurable too.

**Review by ybnd — COMMENTED** [review]
I think this has a lot of potential as a more generic plugin -- some random thoughts:
* Instead of only focusing on "weak" matches, the plugin could map recommendation levels to different commands. 
* These commands don't have to be limited to notifications
* The prompt option to create a "marker file" could also be a configurable command!
  ```sh
  xargs -I $ touch $/beets-unmatched
  ```

**Comment by doronbehar** [comment]
Hey and thanks for the review @ybnd!

Your points have made me think for a while about the purpose of this plugin, and I think I've reached good conclusions. In any case, I'd like to hear about use cases for the ideas you've suggested, in particular:

> - These commands don't have to be limited to notifications
> - The prompt option to create a "marker file" could also be a configurable command!

Although I am interested in hearing ideas, I tend to not want to add to many moving parts to this plugin. In fact, I was thinking even further, that perhaps what's implemented here could become an addition to the `mbsubmit` plugin, instead of a whole new plugin. Because all it tries to do is help making bulk imports easier. I'd be happy to get feedback for the following idea :):

Add another `PromptChoice` for `mbsubmit.py`, that will print the tracks _into a **F**ile_ in the same format as the regular printing. The file name indeed should be configurable, as pointed out by @ybnd. This brings us sort of 1 step closer to resolve #1866.

Then, a user will be able to run something like:

```
picard $(dirname **/beets-unmatched)
```

Which will open `picard` with all of those unmatched directories, making it easy to bulk submitting those items. And also, the `beets-unmatched` file will have useful contents, which perhaps other usages will be found by users.

Eventually though, copying the [`addrelease.py` logic from picard](https://github.com/metabrainz/picard-plugins/blob/2.0/plugins/addrelease/addrelease.py?rgh-link-date=2020-03-15T09%3A49%3A41Z) I mentioned in [#1866](https://github.com/beetbox/beets/issues/1866#issuecomment-599188611), into beets will be the ideal improvement to `mbsubmit` plugin.

What do you think @sampsyo? Perhaps I should open a new PR with at least that idea added to `mbsubmit.py`?

**Comment by sampsyo** [comment]
I'm sorry that I don't have the bandwidth to carefully review this, but on principle, I certainly have no objection to adding that "file export" option to the `mbsubmit` plugin!

**Comment by doronbehar** [comment]
Closing this, as progress seems better at https://github.com/beetbox/beets/pull/4807 .

**Pull request closed without merging**